### PR TITLE
[enhancement]Optimize GeoFunctions for const columns

### PR DESCRIPTION
**Pull request by koarz**
## Proposed changes

Issue Number: #34131 

1000w row data
st_point()
```
before avg: 4.36s
now avg:4.338s
```
st_azimuth()
```
before avg:4.215s
now avg:3.718s
```

st_contains()
```
before avg:4.22s
now avg:3.8275s
```
test sql
```
set parallel_pipeline_task_num=1;
pager cat > /dev/null

select st_point(1.0,a2) from const_double;
select st_azimuth(st_point(1.0,0.0),a2) from geo;
select st_contains(ST_Polygon("POLYGON ((0 0, 10 0, 10 10, 0 10, 0 0))"), a2) from geo;
```
## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by koarz** [comment]
run buildall

File: be/src/vec/functions/functions_geo.cpp [hunk 1]
```diff
@@ -44,20 +45,47 @@ struct StPoint {
...
+        const auto size = std::max(left_column->size(), right_column->size());
 
         MutableColumnPtr res = return_type->create_column();
 
+        if (left_const) {
+            const_vector(left_column, right_column, res, size);
+        } else if (right_const) {
+            vector_const(left_column, right_column, res, size);
+        } else {
+            GeoPoint point;
```
Comment (zclllyybb) [inline be/src/vec/functions/functions_geo.cpp:62]: also wrap this in `vector_vector`

File: be/src/vec/functions/functions_geo.cpp [hunk 2]
```diff
@@ -463,17 +572,76 @@ struct StContains {
...
 
-        const auto size = shape1->size();
         MutableColumnPtr res = return_type->create_column();
 
+        if (left_const) {
+            const_vector(left_column, right_column, res, size);
+        } else if (right_const) {
+            vector_const(left_column, right_column, res, size);
+        } else {
+            int i;
```
Comment (zclllyybb) [inline be/src/vec/functions/functions_geo.cpp:589]: ditto.

File: be/src/vec/functions/functions_geo.cpp [hunk 3]
```diff
@@ -333,8 +416,34 @@ struct StAzimuth {
...
+
+    static void vector_const(const ColumnPtr& left_column, const ColumnPtr& right_column,
+                             MutableColumnPtr& res, size_t size) {
+        GeoPoint point1;
+        GeoPoint point2;
+
+        auto shape_value2 = right_column->get_data_at(0);
+        auto pt2 = point2.decode_from(shape_value2.data, shape_value2.size);
+        for (int row = 0; row < size; ++row) {
+            auto shape_value1 = left_column->get_data_at(0);
```
Comment (zclllyybb) [inline be/src/vec/functions/functions_geo.cpp:429]: for things we did in loop. wrapping them in a function is better

**Review by zclllyybb — CHANGES_REQUESTED** [review]
please add be-ut for Consted calls

**Comment by koarz** [comment]
run buildall

**Comment by koarz** [comment]
run buildall

**Comment by koarz** [comment]
run buildall

**Comment by koarz** [comment]
run buildall

**Comment by koarz** [comment]
run buildall

**Review by zclllyybb — APPROVED** [review]
LGTM

**Review by HappenLee — APPROVED** [review]
LGTM

**Comment by hello-stephen** [comment]
run performance

**Pull request merged**